tooling/ssa_fuzzer/fuzzer/src/fuzz_lib/fuzzer.rs

```diff
@@ -70,96 +70,69 @@ impl Fuzzer {
+            ));
         }
+        Self { contexts }
     }
 
     pub(crate) fn process_function(&mut self, function_data: FunctionData, types: Vec<ValueType>) {
-        if let Some(context) = &mut self.context_non_constant {
+        for context in &mut self.contexts {
             context.process_function(function_data.clone(), types.clone());
         }
-        if let Some(context) = &mut self.context_non_constant_with_idempotent_morphing {
-            context.process_function(function_data.clone(), types.clone());
-        }
-        if let Some(context) = &mut self.context_constant {
-            context.process_function(function_data, types);
-        }
     }
 
-    /// Finalizes the function for both contexts, executes and compares the results
+    /// Finalizes the function for contexts, executes and compares the results
     pub(crate) fn finalize_and_run(
-        mut self,
+        self,
         initial_witness: WitnessMap<FieldElement>,
     ) -> Option<FuzzerOutput> {
-        let mut non_constant_context = self.context_non_constant.take().unwrap();
-        non_constant_context.finalize_program();
-        let non_constant_result =
-            Self::execute_and_compare(non_constant_context, initial_witness.clone());
-
-        if let Some(context) = self.context_constant.take() {
-            let mut constant_context = context;
-            constant_context.finalize_program();
-            let constant_result =
-                Self::execute_and_compare(constant_context, initial_witness.clone());
-            if non_constant_result.is_some() {
-                assert_eq!(
-                    non_constant_result.as_ref().unwrap().get_return_value(),
-                    constant_result?.get_return_value(),
-                    "Non-constant and constant contexts should return the same result"
-                );
-            }
+        let mut execution_results: HashMap<FuzzerMode, Option<FuzzerOutput>> = HashMap::new();
+        for mut context in self.contexts {
+            context.finalize_program();
+            execution_results.insert(
+                context.get_mode(),
+                Self::execute_and_compare(context, initial_witness.clone()),
+            );
         }
-
-        if let Some(context) = self.context_non_constant_with_idempotent_morphing.take() {
-            let mut context_with_idempotent_morphing = context;
-            context_with_idempotent_morphing.finalize_program();
-            let result_with_constrains =
-                Self::execute_and_compare(context_with_idempotent_morphing, initial_witness);
-            if non_constant_result.is_some() {
-                assert_eq!(
-                    non_constant_result.as_ref().unwrap().get_return_value(),
-                    result_with_constrains?.get_return_value(),
-                    "Non-constant and idempotent morphing contexts should return the same result"
-                );
+        for k1 in execution_results.keys() {
```

Comment:
I think it's better to see if everyone has the same result and then to print all results if there is a t least one difference